lib/rules/sort-invocations.js

```diff
@@ -0,0 +1,315 @@
+    if (compareAttributes(attributes[i], attributes[i + 1]) === 1) {
+      canSkip = false;
+
+      break;
+    }
+  }
+
+  return canSkip;
+}
+
+function canSkipSortHash(hash) {
+  let canSkip = true;
+
+  for (let i = 0; i < hash.pairs.length - 1; i++) {
+    if (compareHashPairs(hash.pairs[i], hash.pairs[i + 1]) === 1) {
+      canSkip = false;
+
+      break;
+    }
+  }
+
+  return canSkip;
+}
+
+function canSkipSortModifiers(modifiers) {
+  let canSkip = true;
+
+  for (let i = 0; i < modifiers.length - 1; i++) {
+    if (compareModifiers(modifiers[i], modifiers[i + 1]) === 1) {
+      canSkip = false;
+
+      break;
+    }
+  }
+
+  return canSkip;
+}
+
+function sortAttributes(attributes) {
+  return attributes.sort(compareAttributes).map((attribute) => {
+    const { name, value } = attribute;
+
+    switch (value.type) {
+      case 'ConcatStatement': {
+        // Bug in ember-template-recast@6.1.5 (it removes the single character before or after a MustacheStatement)
+        if (value.parts.length < 2) {
+          break;
+        }
+
+        const firstPart = value.parts[0];
+        const newFirstPart = cloneIfTextNode(firstPart);
+
+        // eslint-disable-next-line unicorn/prefer-at
+        const lastPart = value.parts[value.parts.length - 1];
+        const newLastPart = cloneIfTextNode(lastPart);
+
+        return b.attr(
+          name,
+          // eslint-disable-next-line unicorn/prefer-negative-index, unicorn/prefer-spread
+          b.concat([newFirstPart, ...value.parts.slice(1, value.parts.length - 1), newLastPart])
+        );
+      }
+
+      case 'TextNode': {
+        // Bug in ember-template-recast@6.1.5 (it removes values that are an empty string)
+        if (value.chars === '') {
+          const { start, end } = value.loc;
+
+          const isValueUndefined = start.line === end.line && start.column === end.column;
+
+          if (!isValueUndefined) {
+            return b.attr(name, b.mustache(b.string('')));
+          }
+        }
+
+        break;
+      }
+    }
+
+    return b.attr(name, value);
+  });
+}
+
+function sortHash(hash) {
+  return b.hash(hash.pairs.sort(compareHashPairs));
+}
+
+function sortModifiers(modifiers) {
+  return modifiers.sort(compareModifiers).map((modifier) => {
+    const { hash, params, path } = modifier;
+
+    return b.elementModifier(path, params, hash);
+  });
+}
+
+export default class SortInvocations extends Rule {
+  visitor() {
+    return {
+      BlockStatement(node) {
+        const { hash, params, path } = node;
+
+        if (canSkipSortHash(hash)) {
+          return;
+        }
+
+        const sorted = sortHash(hash);
+
+        if (this.mode === 'fix') {
+          node = b.block(path, params, sorted, b.blockItself());
+        } else {
+          this.log({
+            isFixable: true,
+            message: `Hash keys must be sorted: ${sorted.pairs.map(getHashPairName).join(', ')}`,
+            node,
```

Comment:
I think we may scary user with whole `node` in RED_UNDERLINE (in editor), I'm proposing to log fist incorrectly sorted node here.

Reply:
Done.